types/webpack-assets-manifest/index.d.ts

```diff
@@ -5,98 +5,183 @@
+    maybeMerge(): void;
+
+    /** Emit the assets manifest */
+    emitAssetsManifest(compilation: Compilation): Promise<void>;
+
+    /** Record details of Asset Modules */
+    handleProcessAssetsAnalyse(compilation: Compilation): void;
+
+    /**
+     * When using webpack 5 persistent cache, loaderContext.emitFile sometimes doesn't
+     * get called and so the asset names are not recorded. To work around this, lets
+     * loops over the stats.assets and record the asset names.
+     */
+    processStatsAssets(assets: ReadonlyArray<Asset>): void;
+
+    /** Get assets and hot module replacement files from a compilation object */
+    getCompilationAssets(compilation: Compilation): { assets: Asset[]; hmrFiles: Set<string> };
+
+    /** Gather asset details */
+    handleProcessAssetsReport(compilation: Compilation): Promise<void>;
+
+    /** Get the parsed output path. [hash] is supported. */
+    getManifestPath(compilation: Compilation, filename: string): string;
+
+    /** Write the asset manifest to the file system */
+    writeTo(destination: string): Promise<void>;
+
+    clear(): void;
+
+    /** Cleanup before running Webpack */
+    handleWatchRun(): void;
+
+    /** Determine if the manifest should be written to disk with fs */
+    shouldWriteToDisk(compilation: Compilation): boolean;
+
+    /** Last chance to write the manifest to disk */
+    handleAfterEmit(compilation: Compilation): Promise<void>;
+
+    /** Record asset names */
+    handleNormalModuleLoader(compilation: Compilation, loaderContext: LoaderContext<unknown>, module: Module): void;
+
+    /** Add the SRI hash to the assetsInfo map */
+    recordSubresourceIntegrity(compilation: Compilation): void;
+
+    /** Hook into compilation objects */
+    handleCompilation(compilation: Compilation): void;
+
+    /** Hook into the compilation object */
+    handleThisCompilation(compilation: Compilation): void;
+
+    /**
+     * Determine if webpack-dev-server is being used
+     *
+     * The WEBPACK_DEV_SERVER / WEBPACK_SERVE env vars cannot be relied upon.
+     * See issue {@link https://github.com/webdeveric/webpack-assets-manifest/issues/125|#125}
+     */
+    inDevServer(): boolean;
+
     /** Get the file system path to the manifest */
     getOutputPath(): string;
 
     /** Get the public path for the filename */
     getPublicPath(filename: string): string;
 
     /**
-     * Get a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler) for the manifest
+     * Get a {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler|Proxy} for the manifest.
+     * This allows you to use `[]` to manage entries.
      *
-     * @param raw - Use `setRaw` instead of `set`
+     * @param raw - Should the proxy use `setRaw` instead of `set`?
      */
-    getProxy(raw?: boolean): ProxyHandler<WebpackAssetsManifest>;
+    getProxy(raw?: boolean): WebpackAssetsManifest.Assets;
```

Comment:
`ProxyHandler` was the wrong type, but I'm not sure if this is exactly the right type either. The way the proxy is set up, enumerating over the keys of this value will enumerate over the keys of `WebpackAssetsManifest`.

Reply:
I think it's fine too